examples/micro-ui/micro-ui.c

```diff
@@ -0,0 +1,239 @@
+    {
+        glMatrixMode(GL_MODELVIEW);
+        glPushMatrix();
+        glRotatef(box_rot, 0,1,0);
+        glRotatef(box_rot*0.3f, 1,0,0);
+
+        if(capture_cube) {
+            glViewport(
+                cube_win_rect.x, 
+                display_get_height() - cube_win_rect.y - cube_win_rect.h,
+                cube_win_rect.w, 
+                cube_win_rect.h - 24
+            );
+        } else {
+            glViewport(0,0, display_get_width(), display_get_height());
+        }
+        
+        glCallList(dpl_cube);
+        glPopMatrix();
+    }
+}
+
+int main(void)
+{
+    debug_init_isviewer();
+	debug_init_usblog();
+    
+    joypad_init();
+    dfs_init(DFS_DEFAULT_LOCATION);
+    display_init(RESOLUTION_320x240, DEPTH_16_BPP, 3, GAMMA_NONE, FILTERS_RESAMPLE_ANTIALIAS_DEDITHER);
+    rdpq_init();
+    gl_init();
+    
+    #ifdef DEBUG_RDP
+        rdpq_debug_start();
+    #endif
+
+    surface_t surface = surface_alloc(FMT_RGBA16, display_get_width(), display_get_height());
+
+    rainbow(hue, &clear_color[0], &clear_color[1], &clear_color[2]);
+    dpl_cube = create_cube_dpl();
+
+    // Step 1/5: Make sure you have a small font loaded
+    rdpq_font_t *font = rdpq_font_load("rom:/VCR_OSD_MONO.font64");
+    uint8_t font_id = 1;
+    rdpq_text_register_font(font_id, font);
+
+    // Step 2/5: init UI libary, pass in the controller (joystick or N64-mouse) and font-id
+    // (Note: take a look inside this function for styling and controls.)
+    mu64_init(JOYPAD_PORT_1, font_id);
+
+    float aspect_ratio = (float)display_get_width() / (float)display_get_height();
+    float near_plane = 0.5f;
+    float far_plane = 50.0f;
+
+    glMatrixMode(GL_PROJECTION);
+    glLoadIdentity();
+    glFrustum(-near_plane*aspect_ratio, near_plane*aspect_ratio, -near_plane, near_plane, near_plane, far_plane);
+
+    glMatrixMode(GL_MODELVIEW);
+    glLoadIdentity();
+    gluLookAt(
+        0, 1.8f, 1.8f,
+        0, 0, 0,
+        0, 1, 0
+    );
+
+    static const GLfloat env_color[] = {1,1,1,1};
+    glLightModelfv(GL_LIGHT_MODEL_AMBIENT, env_color);
+
+    glDisable(GL_LIGHTING);
+    glEnable(GL_NORMALIZE);
+    glEnable(GL_DEPTH_TEST);
+    glEnable(GL_CULL_FACE);
+
+    for(;;)
+    {
+        // Game logic
+        joypad_poll();
+
+        uint32_t now_ms = get_ticks_ms();
+        time_delta_ms = now_ms - last_time_ms;
+        last_time_ms = now_ms;
+
+        mu64_start_frame();// Step 3/5: call this BEFORE your game logic starts each frame
+        game_update();
+        mu64_end_frame(); // Step 4/5: call this AFTER your game logic ends each frame
+
+        // Game renderer
+        rdpq_attach(display_get(), &surface);
+
+        gl_context_begin();
+        glClearColor(clear_color[0], clear_color[1], clear_color[2], 1);
+        glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+
+        game_draw(); // (your games renderer)
+
+        mu64_draw(); // Step 5/5: render out the UI at the very end
```

Comment:
Given that this uses rdpq rather than OpenGL, shouldn't it go after `gl_context_end()`?